### PR TITLE
Bug 1506978 - Apply proxy settings to running APBs

### PR DESCRIPTION
**Pull request by eriknelson**
* Broker should inspect its own enviornment, and in the presence of
a proxy configuration, should apply those settings to executing APBs.

### CODE REVIEW

**Comment by eriknelson** [comment]
Currently trying to test this in the example locked down environment, but I confirmed the settings are being applied to running APBs as expected when the various env vars are set.

File: pkg/apb/executor.go [hunk 1]
```diff
@@ -154,6 +140,34 @@ func ExecuteApb(action string,
...
+// GetProxyConfig - Returns a ProxyConfig based on the presence of a proxy
+// configuration in the broker's environment. HTTP_PROXY, HTTPS_PROXY, and
+// NO_PROXY are the relevant environment variables. If no proxy is found,
+// GetProxyConfig will return nil.
+func GetProxyConfig() *ProxyConfig {
+	httpProxy, httpProxyPresent := os.LookupEnv(httpProxyEnvVar)
+	httpsProxy, httpsProxyPresent := os.LookupEnv(httpsProxyEnvVar)
+	noProxy, noProxyPresent := os.LookupEnv(noProxyEnvVar)
+
+	// TODO: Probably some more permutations of these that should be validated?
```
[the remark this answers is not in the document]
Comment (eriknelson) [reply pkg/apb/executor.go]: Yep. I can file an issue after this merges and follow up with a patch + tests.

File: pkg/apb/executor.go [hunk 2]
```diff
@@ -120,24 +123,7 @@ func ExecuteApb(action string,
...
-						v1.EnvVar{
-							Name: "POD_NAMESPACE",
-							ValueFrom: &v1.EnvVarSource{
-								FieldRef: &v1.ObjectFieldSelector{
-									FieldPath: "metadata.namespace",
-								},
-							},
-						},
-					},
+					Env:             createPodEnv(executionContext),
```
Comment (jmrodri) [inline pkg/apb/executor.go]: Nice cleanup!

File: pkg/apb/executor.go [hunk 3]
```diff
@@ -154,6 +140,34 @@ func ExecuteApb(action string,
...
+		return nil
+	}
+
+	return &ProxyConfig{
+		HttpProxy:  httpProxy,
+		HttpsProxy: httpsProxy,
+		NoProxy:    noProxy,
+	}
+}
+
```
Comment (jmrodri) [inline pkg/apb/executor.go]: 👍

File: pkg/apb/executor.go [hunk 4]
```diff
@@ -201,6 +215,52 @@ func createExtraVars(context *Context, parameters *Parameters) (string, error) {
 	return string(extraVars), err
 }
 
+func createPodEnv(executionContext ExecutionContext) []v1.EnvVar {
```
Comment (jmrodri) [inline pkg/apb/executor.go]: EXCELLENT!

File: pkg/apb/executor.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (eriknelson) [reply pkg/apb/executor.go]: #663

**Pull request merged**